v3/mqtt/mqtt.go

```diff
@@ -0,0 +1,49 @@
+package mqtt
+
+import (
+	"errors"
+
+	"github.com/spinframework/spin-go-sdk/v3/internal/fermyon/spin/v2.0.0/mqtt"
+	"go.bytecodealliance.org/cm"
+)
+
+// OpenConnection initializes an MQTT connection
+func OpenConnection(address, username, password string, keepAliveIntervalInSecs uint64) (mqtt.Connection, error) {
+	conn, err, isErr := mqtt.ConnectionOpen(address, username, password, keepAliveIntervalInSecs).Result()
+	if isErr {
+		return cm.ResourceNone, toError(&err)
+	}
+
+	return conn, nil
+}
+
+// Publish publishes an MQTT message
+func Publish(conn mqtt.Connection, topic string, payload []byte, qos uint8) error {
+	_, err, isErr := conn.Publish(topic, mqtt.Payload(cm.ToList(payload)), toQos(qos)).Result()
+	if isErr {
+		return toError(&err)
+	}
+
+	return nil
+}
+
+func toQos(q uint8) mqtt.Qos {
+	switch q {
+	case 0:
+		return mqtt.QosAtMostOnce
+	case 1:
+		return mqtt.QosAtLeastOnce
+	case 2:
+		return mqtt.QosExactlyOnce
+	default:
+		panic("invalid QoS")
+	}
+}
+
+func toError(err *mqtt.Error) error {
```

Comment:
Error messages that can contain strings need to be handled. I'm not sure the cleanest way yet but this is how I implemented it for sqlite https://github.com/spinframework/spin-go-sdk/pull/25/files#diff-4d715145ab0f1dedd50fa907c7bc7b65a6881590b18553e9ccd604f660b240a7R251-R253

There's two errors that can have strings https://github.com/spinframework/spin-go-sdk/blob/wasip2/v3/wit/deps/spin%402.0.0/mqtt.wit#L9-L11